metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -783,7 +788,8 @@ void handleNodeDeactivated(int brokerId, List<ApiMessageAndVersion> records) {
                 record.setIsr(Replicas.toList(newIsr));
                 if (partition.leader == brokerId) {
                     // The fenced node will no longer be the leader.
-                    int newLeader = bestLeader(partition.replicas, newIsr, false);
+                    int newLeader = bestLeader(partition.replicas, newIsr,
```

Comment:
Hmm, how do we prevent the deacitved broker from being selected as the new broker? It seems that at this point, clusterControl hasn't reflected the fenceBrokerRecord yet.

Also, in the caller handleBrokerFenced(), we add the partitionChangeRecord before the fenceBrokerRecord. Ordering wise, it seems that it's more natural to add the fenceBrokerRecord first. Then, it's clear that the partitionChangeRecord is the result of the fenceBrokerRecord.